src/geode/georepresentation/msh_input.cpp

```diff
@@ -514,22 +570,209 @@ namespace
+            mesh_builder.set_edge_vertex(
+                line.mesh().edges_around_vertex( old_line_vertex_id )[0],
+                new_line_vertex_id );
+        }
+
+        void update_component_vertex( const geode::Surface3D& surface,
+            geode::PolygonalSurfaceBuilder3D& mesh_builder,
+            geode::index_t old_surface_vertex_id,
+            geode::index_t new_surface_vertex_id )
+        {
+            OPENGEODE_ASSERT(
+                surface.mesh()
+                        .polygons_around_vertex( old_surface_vertex_id )
+                        .size()
+                    == 1,
+                "By construction, there should be one and only one "
+                "polygon pointing to each vertex at this point." );
+            mesh_builder.set_polygon_vertex(
+                surface.mesh().polygons_around_vertex(
+                    old_surface_vertex_id )[0],
+                new_surface_vertex_id );
+        }
+
+        template < typename Component, typename ComponentMeshBuilder >
+        void filter_dupplicated_vertices( const Component& component,
+            geode::BRep& brep,
+            const ComponentMeshBuilder& mesh_builder )
+        {
+            std::unordered_map< geode::index_t, std::vector< geode::index_t > >
+                unique2component;
+            for( auto v : geode::Range{ component.mesh().nb_vertices() } )
+            {
+                unique2component[brep.unique_vertices().unique_vertex(
+                                     { component.component_id(), v } )]
+                    .emplace_back( v );
+            }
+            geode::BRepBuilder builder{ brep };
+            std::vector< bool > delete_dupplicated(
+                component.mesh().nb_vertices(), false );
+            for( auto uv : unique2component )
+            {
+                for( auto i : geode::Range{ 1, uv.second.size() } )
+                {
+                    update_component_vertex(
+                        component, *mesh_builder, uv.second[i], uv.second[0] );
+                    delete_dupplicated[uv.second[i]] = true;
+                }
+            }
+
+            std::vector< geode::index_t > updated_unique2component;
+            updated_unique2component.reserve( std::count(
+                delete_dupplicated.begin(), delete_dupplicated.end(), false ) );
+            for( auto i : geode::Range{ component.mesh().nb_vertices() } )
+            {
+                if( delete_dupplicated[i] )
+                {
+                    continue;
+                }
+                auto ui = brep_.unique_vertices().unique_vertex(
+                    { component.component_id(), i } );
+                updated_unique2component.push_back( ui );
+            }
+
+            mesh_builder->delete_vertices( delete_dupplicated );
+
+            builder.unique_vertices().remove_component( component );
+            builder.unique_vertices().register_component( component );
```

Comment:
What are the goal of these two lines ?

Reply:
At this point, component mesh vertices are used once (duplicated mesh element vertices), and vertex identifier was used during file reading and loading. We want to merge common vertices (those pointing to the same vertex identifier unique vertices). So I clear the unique_vertices() to update ids by removing et re-registering the component. I can put code block between l.699 to l.707 in a single function named update_unique_vertices() or something like that. What do you think of this?